File: ClientApp/src/app/models/variable.ts
import { Subgroup } from "./group";

export enum VariableType {
  None,
  Bool,
  Byte,
  Word,
  Dword,
  Real,
  String
}

export class Variable {
  constructor(
    public id: number,
    public address: number,
    public type: VariableType,
    public name: string,
    public value: object,
    public active: number,
    public trendingPeriod:number
  ) { }
}

export class VariableEntity {
  constructor(
    public variable: Variable,
    public subgroup: Subgroup,
    public writable: boolean
  ) { }

}

File: Models/Variable.cs
namespace Project1.Models
{
    public enum VariableType
    {
        None,
        Bool,
        Byte,
        Word,
        Dword,
        Real,
        String
    }

    public class Variable
    {
        public int? Id { get; set; }
        public int Address { get; set; }
        public VariableType Type { get; set; }
        public string Name { get; set; } = "";
        public object? Value { get; set; }    
        public int Active { get; set; }
        public int? TrendingPeriod { get; set; }
    }
}
